blog/_posts/2018-05-02-developing-on-kubernetes.md

```diff
@@ -0,0 +1,366 @@
+
+Now that you have a basic idea of the options around the runtime environment, let’s move on to how to iteratively develop and deploy your app.
+
+## The tools of the trade
+
+We are now going to review tooling allowing you to develop apps on Kubernetes with the focus on having minimal impact on your existing workflow. We strive to provide an unbiased description including implications of using each of the tools in general terms.
+
+Note that this is a tricky area since even for established technologies such as, for example, JSON vs YAML vs XML or REST vs gRPC vs SOAP a lot depends on your background, your preferences and organizational settings. It’s even harder to compare tooling in the Kubernetes ecosystem as things evolve very rapidly and new tools are announced almost on a weekly basis; during the preparation of this post alone, for example, [Gitkube](https://gitkube.sh/) and [Watchpod](https://github.com/MinikubeAddon/watchpod) came out. To cover these new tools as well as related, existing tooling such as Weave Flux and OpenShift’s [S2I](https://docs.openshift.com/container-platform/3.9/creating_images/s2i.html) we are planning a follow-up blog post to the one you’re reading.
+
+### Draft
+
+[Draft](https://github.com/Azure/draft) aims to help you get started deploying any app to Kubernetes. It is capable of applying heuristics as to what programming language your app is written in and generates a Dockerfile along with a Helm chart. It then runs the build for you and deploys resulting image to the target cluster via the Helm chart. It also allows user to setup port forwarding to localhost very easily.
+
+Implications:
+* User can provide customise the chart and Dockerfile templates however they like, or even provide custom templates for future use
+* It’s not very simple to guess how just any app is supposed to be built, in some cases user may need to tweak Dockerfile and the Helm chart that Draft generates
+* With [Draft version 0.11.0](https://github.com/Azure/draft/releases/tag/v0.12.0) or older, every time user wants to test a change, they need to wait for Draft to copy the code to the cluster, then run the build, push the image and release updated chart; this can timely, but it results in an image being for every single change made by the user (whether it was committed to git or not)
+* As of Draft version 0.12.0, builds are executed locally
+* User doesn’t have an option to choose something other than Helm for deployment
+* It can watch local changes and trigger deployments, but this is feature is not enabled by default
+* It allows developer to use either local or remote Kubernetes cluster
+* Deploying to production is up to the user, Draft authors recommend their other project – Brigade
+* Can be used instead of Skaffold, and along the side of Squash
```

Comment:
provide links to Skaffold and Squash?

Reply:
I don't think that makes a lot of sense since both Skaffold and Squash have their own sections in the blog post (incl. links).

Reply:
It's in the text already, seems unnecessary...

Reply:
okay